File: quote.js
const Discord = require("discord.js");
const fetch = require("node-fetch")

const client = new Discord.Client();

function getQuote() {
  return fetch("https://zenquotes.io/api/random")
  .then(res => {
    return res.json()
  })
  .then(data => {
    return data[0]["q"] + " -" + data[0]["a"]
  })
}

client.on("ready", ()=> {
  console.log(`logged in as ${client.user.tag}`);
});

client.on("message", msg => {
  if(msg.author.bot) return
  else if(msg.content.includes("qhote")){
    msg.channel.send("use $quote)")
  }
  else if(msg.content === "$quote"){
    getQuote()
    .then(quote => msg.channel.send(quote))
  }

  if(msg.content.startsWith("$creategame"))
});

client.login(process.env.TOKEN);

File: go/bot/quote.go
package bot

import (
	"encoding/json"
	"fmt"
	"io"
	"net/http"
)

type Quote struct {
	Q string `json:"q"`
	A string `json:"a"`
}

func GetQuote() string {
	//Our API returns a quote in an array, so in order to get the actual object, we need to put it in a slice
	var quote []Quote
	res, _ := http.Get("https://zenquotes.io/api/random")
	data, _ := io.ReadAll(res.Body)
	json.Unmarshal(data, &quote)
	return fmt.Sprintf("\"%s\"\n-%s", quote[0].Q, quote[0].A)

}
